boot/bootutil/src/loader.c

```diff
@@ -146,7 +146,8 @@ boot_add_shared_data(struct boot_loader_state *state,
 
 #ifdef MCUBOOT_DATA_SHARING
     rc = boot_save_shared_data(boot_img_hdr(state, active_slot),
-                                BOOT_IMG_AREA(state, active_slot));
+                                BOOT_IMG_AREA(state, active_slot),
+                                active_slot);
```

Comment:
I am not sure whether this is correct. I mean sharing `active_slot` may be correct because this is MCUboot/MCUmgr established slot numbering, but the BOOT_IMG_AREA(state, active_slot) pointer (`struct flash_area`) may differ between MCUboot and app builds, so sharing it does not help.

Reply:
`active_slot` is what I added for XIP, the boot image area is part of the original API which presumably things like mynewt use so don't want to change/break it.